pkg/execution/state/redis_state/queue.go

```diff
@@ -1638,8 +1646,41 @@ func (q *queue) Peek(ctx context.Context, partition *QueuePartition, until time.
 	)
 }
 
+func (q *queue) PeekRandom(ctx context.Context, partition *QueuePartition, until time.Time, limit int64) ([]*osqueue.QueueItem, error) {
+	ctx = redis_telemetry.WithScope(redis_telemetry.WithOpName(ctx, "Peek"), redis_telemetry.ScopeQueue)
+	if q.primaryQueueShard.Kind != string(enums.QueueShardKindRedis) {
+		return nil, fmt.Errorf("unsupported queue shard kind for Peek: %s", q.primaryQueueShard.Kind)
+	}
+	if partition == nil {
+		return nil, fmt.Errorf("expected partition to be set")
+	}
+	if limit > AbsoluteQueuePeekMax {
+		// Lua's max unpack() length is 8000; don't allow users to peek more than
+		// 1k at a time regardless.
+		limit = AbsoluteQueuePeekMax
+	}
+	if limit > q.peekMax {
+		limit = q.peekMax
+	}
+	if limit <= 0 {
+		limit = q.peekMin
+	}
+	partitionKey := partition.zsetKey(q.primaryQueueShard.RedisClient.kg)
+	return q.peek(
+		ctx,
+		q.primaryQueueShard,
+		peekOpts{
+			Limit:        limit,
+			Until:        until,
+			PartitionKey: partitionKey,
+			Random:       true,
```

Comment:
could this be made an configuration option?
so it's possible to enable it selectively without impacting everyone at once.